lib/SILOptimizer/Mandatory/SendNonSendable.cpp

```diff
@@ -531,94 +616,171 @@ class PartitionOpTranslator {
+    // these are treated as stores - meaning that they could write values into
+    // memory. The beahvior of this depends on whether the tgt addr is aliased,
+    // but conservative behavior is to treat these as merges of the regions
+    // of the src value and tgt addr
     if (isa<CopyAddrInst,
             ExplicitCopyAddrInst,
             StoreInst,
             StoreBorrowInst,
-            StoreWeakInst>(instruction)) {
+            StoreWeakInst>(instruction))
       return translateSILStore(
           instruction->getOperand(1),
           instruction->getOperand(0));
-    }
 
-    // Handle applications
-    if (isApplyInst(*instruction)) {
+    // handle applications
+    if (isApplyInst(*instruction))
       return translateSILApply(instruction);
-    }
 
-    // handle tuple destruction and creation
-    if (auto destructTupleInst = dyn_cast<DestructureTupleInst>(instruction)) {
+    // handle tuple destruction
+    if (auto destructTupleInst = dyn_cast<DestructureTupleInst>(instruction))
       return translateSILMultiAssign(
           {destructTupleInst->getResults().begin(),
            destructTupleInst->getResults().end()},
           {destructTupleInst->getOperand()});
-    }
 
-    if (auto tupleInst = dyn_cast<TupleInst>(instruction)) {
+    // handle instructions that aggregate their operands into a single structure
+    // - treated as a multi assign
+    if (isa<ObjectInst,
+            StructInst,
+            TupleInst>(instruction))
       return translateSILMultiAssign(
-          {tupleInst->getResult(0)},
-          {tupleInst->getOperandValues().begin(),
-           tupleInst->getOperandValues().end()});
-    }
-
-    // Handle returns - require the operand to be non-consumed
-    if (auto *returnInst = dyn_cast<ReturnInst>(instruction)) {
-      return translateSILRequire(returnInst->getOperand());
-    }
-
-    if (isa<ClassMethodInst,
+          {instruction->getResult(0)},
+          {instruction->getOperandValues().begin(),
+           instruction->getOperandValues().end()});
+
+    // Handle returns and throws - require the operand to be non-consumed
+    if (isa<ReturnInst, ThrowInst>(instruction))
+      return translateSILRequire(instruction->getOperand(0));
+
+    // handle branching terminators
+    // in particular, need to handle phi-node-like argument passing
+
+    if (auto branchInst = dyn_cast<BranchInst>(instruction))
+      return translateSILPhi({{
+          {branchInst->getArgs().begin(), branchInst->getArgs().end()},
+          branchInst->getDestBB()}});
+
+    if (auto condBranchInst = dyn_cast<CondBranchInst>(instruction))
+      return translateSILPhi({
+          { // true branch
+           {condBranchInst->getTrueArgs().begin(), condBranchInst->getTrueArgs().end()},
+           condBranchInst->getTrueBB()},
+          { // false branch
+           {condBranchInst->getTrueArgs().begin(), condBranchInst->getTrueArgs().end()},
+           condBranchInst->getTrueBB()}});
```

Comment:
This looks a little suspect, since both are referring to the `TrueArgs` and `TrueBB`.